tests/api_test.py

```diff
@@ -947,6 +947,12 @@ def test_jvp_mismatched_arguments(self):
       TypeError,
       "primal and tangent arguments to jax.jvp do not match.",
       lambda: api.jvp(lambda x: -x, (np.float16(2),), (np.float32(4),)))
+    # If primals and tangents are not of the same shape then raise error
+    self.assertRaisesRegex(
```

Comment:
We are trying to transition to the new style of checking for exceptions, using `with self.assertRaisesRegex(...)`.